mobile/src/main/java/com/ataulm/wutson/discover/GenresPagerAdapter.java

```diff
@@ -31,13 +33,26 @@ void update(List<ShowsInGenre> showsInGenres) {
 
     @Override
     public View instantiateItem(ViewGroup container, int position) {
-        DiscoverByGenreView view = (DiscoverByGenreView) layoutInflater.inflate(R.layout.view_discover_by_genre, container, false);
-        ShowsInGenre showsInGenre = showsInGenres.get(position);
-        view.update(showsInGenre, listener);
+        RecyclerView view = createView(container);
```

Comment:
remove unnecessary levels of custom view where "unnecessary" is of course pretty subjective.